### ISSUE TITLE
fs-uae fails to build when building with many parallel jobs

### ISSUE THREAD

**Issue by glaubitz** [FrodeSolheim/fs-uae#222]
On Debian, ```fs-uae``` occasionally fails to build from source when many parallel jobs are used:

```
dh_auto_build
	make -j16
make[2]: Entering directory '/<<PKGBUILDDIR>>'
echo "int cdummy;" > ./gen/cdummy.c
echo "int cppdummy;" > ./gen/cppdummy.cpp
/bin/bash: ./gen/cppdummy.cpp: No such file or directory
make[2]: *** [Makefile:6185: gen/cppdummy.cpp] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/<<PKGBUILDDIR>>'
dh_auto_build: error: make -j16 returned exit code 2
make[1]: *** [debian/rules:8: override_dh_auto_build] Error 25
make[1]: Leaving directory '/<<PKGBUILDDIR>>'
make: *** [debian/rules:29: build-arch] Error 2
dpkg-buildpackage: error: debian/rules build-arch subprocess returned exit status 2
```

Reducing the number of build jobs helps.

Full build log: https://buildd.debian.org/status/fetch.php?pkg=fs-uae&arch=ppc64&ver=3.0.3%2Bdfsg-1&stamp=1585322512&raw=0

**Comment by FrodeSolheim** [comment]
Figured out what the problem is, and also, the gen/cppdummy.cpp (and gen/cdummy.c file) is no longer needed really.

A workaround is to create the [build_dir]/gen directory before starting the build, or just remove the following lines from Makefile.am:

```
diff --git a/Makefile.am b/Makefile.am
index 53365d69..2d9a86c2 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1133,18 +1133,6 @@ CM_CFLAGS += @LIBMPEG2_CFLAGS@
 LIBS += @LIBMPEG2_LIBS@
 endif
 
-noinst_LIBRARIES += libdummy.a
-nodist_libdummy_a_SOURCES = \
-       gen/cdummy.c \
-       gen/cppdummy.cpp
-BUILT_SOURCES += $(nodist_libdummy_a_SOURCES)
-
-gen/cdummy.c:
-       echo "int cdummy;" > $(b)/gen/cdummy.c
-
-gen/cppdummy.cpp:
-       echo "int cppdummy;" > $(b)/gen/cppdummy.cpp
-
 gen_build68k_SOURCES = \
        src/build68k.cpp \
        src/writelog.cpp
```
I'm removing these lines from the git repo. Would you like a new release tarball with this removed to resolve this error, or just patch the debian package?

**Comment by glaubitz** [comment]
A new release would be nice, but that should also fix #224 and include the patch from #219.

This way, fs-uae would be ready for the upcoming gcc-10 and I can also drop the patch for unkown CPUs.

**Comment by FrodeSolheim** [comment]
Fixed in https://github.com/FrodeSolheim/fs-uae/commit/8ce4ed25e4f2eafeec1f3bc7a0530e0ceac3d040

**Comment by FrodeSolheim** [comment]
@glaubitz I've made a 3.0.4 release: https://fs-uae.net/stable/3.0.4/ :)

**Comment by glaubitz** [comment]
Cool, thank you. I'll package it tomorrow :).

**Comment by FrodeSolheim** [comment]
@glaubitz There might be a 3.0.5 soon with a small bugfix, so please wait until the day after tomorrow ;-)

**Comment by glaubitz** [comment]
Ok, thanks. I haven't done anything on Debian yet, but I already submitted 3.0.4 to openSUSE.

Would be great if my small patch to support unknown CPUs would be in 3.0.5 as well then I could submit ```fs-uae``` to Debian without any additional packages which makes the packaging effort easier and faster.

**Comment by FrodeSolheim** [comment]
@glaubitz https://fs-uae.net/stable/3.0.5/ :)

(The unknown CPU patch was actually included in 3.0.4)

FS-UAE
----------

Version 3.0.5:

* Packaging change to satisfy Apple's changed notarization process (sigh).

Version 3.0.4:

* Fix compilation of sana2.cpp with GCC 10.
* Fix OpenGL shaders with multiple passes on macOS.
* Include compilation fix for "unknown" CPUs architectures.
* Updated the GLAD OpenGL loader.

FS-UAE Launcher
------------------------

Version 3.0.5:

* Packaging change to satisfy Apple's changed notarization process (sigh).
* Workaround for old file database entries with incorrect \ escape.

Version 3.0.4:

* Filename escaping for FS-UAE and .uaem comments when using .lha files as HDs.
* Fixed bug in escaping of \ in archive.py.
